Packs/Slack/Integrations/SlackV3/SlackV3.py

```diff
@@ -1417,158 +1382,169 @@ def handle_newly_created_channel(creator, channel):
-        if data.get('command', None):
-            demisto.debug("Slash command event received. Ignoring.")
-            return
+    MESSAGE_QUEUE.put(req)
 
-        # Check to see if the event is about a newly handled event.
-        elif event.get('type') == 'channel_created' and MIRRORING_ENABLED:
-            demisto.debug("Found new channel")
-            creator = event.get('channel', {}).get('creator', '')
-            channel_id = event.get('channel', {}).get('id', '')
-            handle_newly_created_channel(creator=creator, channel=channel_id)
-            return
 
-        # Check if the message is from a bot
-        if is_bot_message(data):
-            return
+def consume_and_process_message(queue: Queue, i: int):
+    """
+    This is the main listener which is attached to the open socket connection. When a SocketModeRequest has been received
+    this flow will be triggered. The Payload can be of any type and this function will determine what type it is, then
+    handle it accordingly. In the event that the request is nothing of interest to us, we will simply acknowledge the
+    request, process the message and gracefully ignore it.
+    :queue: Message Queue containing messages to be processed
+    :i: Identifier for the thread process
+    :return: None
+    """
+    while True:
+        req = queue.get()
+        data_type: str = req.type
+        payload: dict = req.payload
+        if data_type == 'error':
+            error = payload.get('error', {})
+            error_code = error.get('code')
+            error_msg = error.get('msg')
+            handle_listen_error(
+                f'Slack API has thrown an error. Code: {error_code}, Message: {error_msg}.')
+            continue
+        try:
+            data: dict = req.payload
+            event: dict = data.get('event', {})
+            text = event.get('text', '')
+            user_id = data.get('user', {}).get('id', '')
+            if not user_id:
+                user_id = event.get('user', '')
+            channel = event.get('channel', '')
+            thread = event.get('thread_ts', None)
+            message = data.get('message', {})
+            action_text = ''
+            message_ts = message.get('ts', '')
+            actions = data.get('actions', [])
+            state = data.get('state', {})
+
+            # Check if slash command received. If so, ignore for now.
+            if data.get('command', None):
+                demisto.debug("Slash command event received. Ignoring.")
+                continue
 
-        # Sometimes we can receive message_changed events, we currently do not support these.
-        if event.get('subtype') == 'message_changed':
-            demisto.debug("Received message_changed event type. Ignoring.")
-            return
+            # Check to see if the event is about a newly handled event.
+            elif event.get('type') == 'channel_created' and MIRRORING_ENABLED:
+                demisto.debug("Found new channel")
```

Comment:
Better to do it 2 lines below, and add the new channel info (`creator` and `channel_id`.)